build/mac/notarize_dmg_pkg.py

```diff
@@ -0,0 +1,186 @@
+#!/usr/bin/env python
+
+# Copyright (c) 2019 The Brave Authors. All rights reserved.
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import argparse
+import os
+import subprocess
+import sys
+
+"""
+The sign_chrome.py script does notarization of the app, but we don't use it to
+create our .dmg and .pkg files. Instead we do our own signing and packaging,
+but must perform notarization after those processes are complete. This script
+performs notarizing and stapling of those files.
+"""
+
+# Our CWD is the packaging directory (i.e.
+# src/out/Release/Brave_Browser_CHANNEL_Packaging), the signing directory is
+# relative to that
+packaging_signing_path = os.path.realpath(os.path.dirname(os.path.realpath(__file__)))
+sys.path.append(packaging_signing_path)
+
+# Import the entire module to avoid circular dependencies in the functions
+from signing import config, commands, model, notarize, pipeline, signing  # noqa: E402
+
+
+def run_command(args, **kwargs):
+    print('Running command: {}'.format(args))
+    subprocess.check_call(args, **kwargs)
+
+
+def create_config(config_args, development, mac_provisioning_profile):
+    """Creates the |model.CodeSignConfig| for the signing operations.
+
+    If |development| is True, the config will be modified to not require
+    restricted internal assets, nor will the products be required to match
+    specific certificate hashes.
+
+    Args:
+        config_args: List of args to expand to the config class's constructor.
+        development: Boolean indicating whether or not to modify the chosen
+            config for development testing.
+
+    Returns:
+        An instance of |model.CodeSignConfig|.
+    """
+    config_class = config.CodeSignConfig
+    """
+    try:
+        import signing.internal_config
+        config_class = signing.internal_config.InternalCodeSignConfig
+    except ImportError as e:
+        # If the build specified Google Chrome as the product, then the
+        # internal config has to be available.
+        if config_class(identity, keychain).product == 'Google Chrome':
+            raise e
+    """
+
+    if development:
+
+        class DevelopmentCodeSignConfig(config_class):
+
+            @property
+            def codesign_requirements_basic(self):
+                return ''
+
+            @property
+            def provisioning_profile_basename(self):
+                return None
+
+            @property
+            def run_spctl_assess(self):
+                return False
+
+        config_class = DevelopmentCodeSignConfig
+
+    config_class = GetBraveSigningConfig(config_class, development, mac_provisioning_profile)
+    return config_class(*config_args)
+
+
+def GetBraveSigningConfig(config_class, development, mac_provisioning_profile):
```

Comment:
Would it be possible to extend script/signing_helper.py with an extra param and import from there?

Reply:
I attempted this at one point but had import issues that prevented it from working. I just tried extending the function again and it works just fine. Will push a commit shortly that allows both a profile name passed in as an optional argument, as well as the env variable method.